test/unit-tests/adapters/axelar/AxelarReceiverAdapter.t.sol

```diff
@@ -0,0 +1,390 @@
+        address[] memory receiverAdapters = new address[](1);
+        receiverAdapters[0] = address(dummyAdapter);
+        bool[] memory operations = new bool[](1);
+        operations[0] = true;
+        MultiMessageReceiver(receiverAddr).updateReceiverAdapter(receiverAdapters, operations);
+
+        vm.startPrank(caller);
+
+        MessageLibrary.Message memory message = MessageLibrary.Message({
+            srcChainId: SRC_CHAIN_ID,
+            dstChainId: DST_CHAIN_ID,
+            target: address(42),
+            nonce: 0,
+            callData: bytes("42"),
+            nativeValue: 0,
+            expiration: type(uint256).max
+        });
+        bytes32 msgId = MessageLibrary.computeMsgId(message);
+
+        AdapterPayload memory payload = AdapterPayload({
+            msgId: msgId,
+            senderAdapterCaller: address(42),
+            receiverAdapter: address(43),
+            finalDestination: address(44),
+            data: abi.encode(message)
+        });
+
+        vm.expectRevert(Error.INVALID_FINAL_DESTINATION.selector);
+        dummyAdapter.execute(bytes32("commandId"), "ethereum", senderAdapter.toString(), abi.encode(payload));
+    }
+
+    /// @dev reverts if message fails to be received
+    function test_execute_message_failure() public {
+        vm.startPrank(owner);
+
+        address senderAdapter = contractAddress[SRC_CHAIN_ID]["AXELAR_SENDER_ADAPTER"];
+        AxelarReceiverAdapter dummyAdapter =
+            new AxelarReceiverAdapter(address(new MockAxelarGateway(true)), contractAddress[DST_CHAIN_ID]["GAC"]);
+        dummyAdapter.updateSenderAdapter(abi.encode("ethereum"), senderAdapter);
+
+        vm.startPrank(contractAddress[DST_CHAIN_ID]["TIMELOCK"]);
+        address receiverAddr = contractAddress[DST_CHAIN_ID]["MMA_RECEIVER"];
+        address[] memory receiverAdapters = new address[](1);
+        receiverAdapters[0] = address(dummyAdapter);
+        bool[] memory operations = new bool[](1);
+        operations[0] = true;
+        MultiMessageReceiver(receiverAddr).updateReceiverAdapter(receiverAdapters, operations);
+
+        vm.startPrank(caller);
+
+        MessageLibrary.Message memory message = MessageLibrary.Message({
+            srcChainId: SRC_CHAIN_ID,
+            dstChainId: DST_CHAIN_ID,
+            target: address(0), // invalid target
+            nonce: 0,
+            callData: bytes("42"),
+            nativeValue: 0,
+            expiration: type(uint256).max
+        });
+        bytes32 msgId = MessageLibrary.computeMsgId(message);
+
+        AdapterPayload memory payload = AdapterPayload({
+            msgId: msgId,
+            senderAdapterCaller: address(42),
+            receiverAdapter: address(43),
+            finalDestination: receiverAddr,
+            data: abi.encode(message)
+        });
+
+        // NOTE: Forge mangles low level error and doesn't allow checking for partial signature match
+        vm.expectRevert();
```

Comment:
suggest the following to test the correct error:
```suggestion
        vm.expectRevert(abi.encodeWithSelector(MessageExecutor.MessageFailure.selector, msgId, abi.encodePacked(Error.INVALID_TARGET.selector)));
```

Reply:
Done.